### PR TITLE
Filter by GCN Events on the Source Table Form

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR starts to add GcnEvents to the source table filter form.

@Theodlz Do you understand the form mechanics well enough to know how to get the selected IDs out? This form seems different than most, with regards to how the submit is handled.

### CODE REVIEW

**Comment by Theodlz** [comment]
@mcoughlin it's done :) Just committed it to your branch. I'll let you try it.
Note: in your useEffect, you used to set a default gcnEvent. However if you do that you force people to filter by a gcn event, so I removed it. Also; it happens to not work well with a controlled form in general.

**Comment by Theodlz** [comment]
@mcoughlin I took care of the front-end. However, I had a question. For example in the GCN Event page, when we query sources within a localization, we use a start and end date that corresponds to up to 7 days after the event. Here, that would query all the sources within the localization at ALL times. Should I (in the pre-submit function I added) always add a start and end date filtering of one week when a user filters by localizationName and dateobs ?

**Comment by Theodlz** [comment]
It could even be a backend thing. If you filter by localizationName and dateobs, and that you didn't specify a start and end date, then we add one of one week to be sure to limit API requests that filter on those without specifying start and end dates.

**Comment by mcoughlin** [comment]
@Theodlz Great idea. Another option would be to make start and end date required in some form of validation in the form.

**Comment by Theodlz** [comment]
In the handler, what do you think of doing something like:
```
if localization_dateobs is not None or localization_name is not None:
            if saved_after is None or saved_before is None:
                raise self.error(
                    400,
                    "Must specify savedAfter and savedBefore when filtering by localizationDateobs or localizationName",
                )
            # else verify that the period between saved_after and saved_before is less than a month
            if saved_after > saved_before:
                raise self.error(
                    400,
                    "savedAfter must be before savedBefore when filtering by localizationDateobs or localizationName",
                )
            if (saved_before - saved_after).days > 31:
                raise self.error(
                    400,
                    "savedAfter and savedBefore must be less than a month apart when filtering by localizationDateobs or localizationName",
                )
```

**Comment by mcoughlin** [comment]
@Theodlz I like that. I suggest we define the 31 day variable at the top of the file like we do some of the other thresholds.

**Comment by Theodlz** [comment]
Of course. If you have in mind a period that is more significant that 31 days please tell me !

**Comment by mcoughlin** [comment]
@Theodlz We can start there (it feels reasonable), and if it poses difficulty or someone else has another use case we can consider expanding it (the BBH folks might?).

**Comment by Theodlz** [comment]
Hi @mcoughlin , I added the frontend and backend verification we talked about.
Also, I expected some bugs with the source handler when passing a first and last detected date from the frontend, and had to do some modifications that might impact some other handlers that use the get_sources method. Basically, in the get method of the source handler, there is a piece of code that verify the different dates passed as parameters. However, at the same time, it converts them in datetime objects. As they aren't string anymore, the ".strip()" applied to them in the get_sources() method does not exist, and we get an error.
We should decide if: we now pass the dates as datetime objects, or convert them back to strings after verification.

**Comment by mcoughlin** [comment]
@Theodlz I believe you are right about the first option too. The phrases seem unnecessarily complicated as written!

**Review by mcoughlin — COMMENTED** [review]
@Theodlz This is looking great. Maybe class UTCTZnaiveDateTime(fields.DateTime): can be put in its own function in utils for re-use?

**Review by guynir42 — APPROVED** [review]
Looks good. I have a couple of comments.

File: skyportal/tests/api/test_gcn.py
Comment (Theodlz) [reply skyportal/tests/api/test_gcn.py:655]: Yes. First I noticed the test was wrong, I'm adding 9 obs not 10. Plus adding a superior instead of an equal fixes some issues with other tests also adding some obs.

File: static/js/components/SourceTableFilterForm.jsx
Comment (Theodlz) [reply static/js/components/SourceTableFilterForm.jsx]: Done ! What i did on the frontend is that it uses the value coming from the config, but to avoid errors, if this value is not accessible for some reason, it defaults to 31.

Comment (Theodlz) [reply static/js/components/SourceTableFilterForm.jsx]: Done ! What i did on the frontend is that it uses the value coming from the config, but to avoid errors, if this value is not accessible for some reason, it defaults to 31.

File: skyportal/handlers/api/gcn.py
Comment (Theodlz) [reply skyportal/handlers/api/gcn.py:808]: Sounds good. Let me do it now before merging.

Comment (Theodlz) [reply skyportal/handlers/api/gcn.py]: Done.

File: skyportal/handlers/api/config_handler.py
Comment (guynir42) [inline skyportal/handlers/api/config_handler.py:11]: Maybe `MAX_NUM_DAYS_USING_LOCALIZATION` and `maxNumDaysUsingLocalization`

**Review by guynir42 — APPROVED** [review]
LGTM

**Pull request merged**
